azure/storage_account/data.tf

```diff
@@ -0,0 +1,3 @@
+data "azurerm_resource_group" "rg" {
+  name = var.resource_group_name
+}
```

Comment:
[recommendation] I would move the data block to the main file instead, so we can easily follow the flow in one place only.